main.go

```diff
@@ -162,47 +133,105 @@ func main() {
 			return err
 		}
 		return nil
-	}); err != nil {
-		log.Errorf("Nuget restore failed, error: %s", err)
-		os.Exit(1)
+	})
+}
+
+// collectCaches collects the caches based on the allowed config.
+// For more information about caches please read: https://docs.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders
+func collectCaches(allowCaching string) error {
+	nuGetCache := cache.New()
+	switch allowCaching {
+	case cacheInputNone:
+		return nil
+	case cacheInputlocal:
+		collectLocalCaches(&nuGetCache)
+	case cacheInputGlobal:
+		collectGlobalCaches(&nuGetCache)
 	}
+	return nuGetCache.Commit()
+}
 
-	// Collecting caches
-	fmt.Println()
-	log.Infof("Collecting NuGet cache...")
+// collectHTTPCaches collects the HTTP cache.
+func collectHTTPCaches(nuGetCache *cache.Cache) {
+	httpCachePth := getHTTPCachePath()
+	if exists, err := pathutil.IsPathExists(httpCachePth); err != nil {
+		log.Warnf("Failed to determine if path (%s) exists, error: %s", httpCachePth, err)
+	} else if exists {
+		nuGetCache.IncludePath(httpCachePth)
+	}
+}
 
-	nugetCache := cache.New()
+// getHTTPCachePath gets the path for the HTTP cache.
+func getHTTPCachePath() string {
+	if os.Getenv(cacheEnvHTTP) != "" {
+		return os.Getenv(cacheEnvHTTP)
+	}
+	return filepath.Join(pathutil.UserHomeDir(), ".local", "share", "NuGet", "v3-cache")
+}
 
-	xamarinHomeCache := filepath.Join(pathutil.UserHomeDir(), ".local", "share", "Xamarin")
+// collectGlobalCaches collects the global package caches.
+func collectGlobalCaches(nuGetCache *cache.Cache) {
+	nuGetCache.IncludePath(getGlobalCachePath())
+}
 
-	if exists, err := pathutil.IsPathExists(xamarinHomeCache); err != nil {
-		log.Warnf("Failed to determine if path (%s) exists, error: %s", xamarinHomeCache, err)
-	} else if exists {
-		nugetCache.IncludePath(xamarinHomeCache)
+// getGlobalCachePath gets the path for the global cache.
+func getGlobalCachePath() string {
+	if os.Getenv(cacheEnvGlobal) != "" {
+		return os.Getenv(cacheEnvGlobal)
 	}
+	return filepath.Join(pathutil.UserHomeDir(), ".nuget", "packages")
+}
 
+// collectLocalCaches collects the local caches.
+func collectLocalCaches(nuGetCache *cache.Cache) {
```

Comment:
return the errors and let the caller decide how to handle them

Reply:
Updated to return the errors.